wasm/argon2/benchmark.js

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+let passwordStrings = [
+    '123456',
+    'p@assw0rd',
+    'qwerty',
+    '汉字漢字',
+    'كلمة المرور',
+    'Z7ihQxGE93',
+    '0pjTjkrnsM',
+    '6Kg3AmWnVc',
+];
+
+// Emscripten doesn't have a way to manage your pointers for you so these will free them when the wrapper dies.
+const mallocRegistry = new FinalizationRegistry((ptr) => Module._free(ptr));
+
+class CString {
+    constructor(string) {
+        this.ptr = Module.stringToNewUTF8(string);
+        this.length = Module._strlen(this.ptr);
+        mallocRegistry.register(this, this.ptr);
+    }
+}
+
+class MallocPtr {
+    constructor(size) {
+        this.ptr = Module._malloc(size);
+        this.size = size;
+        mallocRegistry.register(this, this.ptr);
+    }
+}
+
+const tCost = 2;
+const mCost = 1024;
+const parallelism = 1;
+const argon2NumberOfTypes = 2;
```

Comment:
Could we make it more self-explanatory what the types are, maybe simply with a comment:
```
// There are three argon2 algorithm variants. We test all three.
// See https://github.com/P-H-C/phc-winner-argon2/blob/f57e61e19229e23c4445b85494dbf7c07de721cb/include/argon2.h#L221. 
```

Reply:
Or if we only test a single variant, we could change this to simply
```
// See https://en.wikipedia.org/wiki/Argon2 for the three algorithm variants.
const argon2idType = 2
```